### PR TITLE
[REF-6194] Fix SageMaker automatic IAM role creation

### PR DESCRIPTION
**Pull request by zohar-mizrahi**

### CODE REVIEW

**Review by yakov-g — APPROVED** [review]

File: mlcomp/parallelm/ml_engine/sagemaker_engine.py [hunk 1]
```diff
@@ -47,29 +50,63 @@ def iam_role(self):
...
+        Creates an IAM role for SageMaker service (
+        https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html#sagemaker-roles-amazonsagemakerfullaccess-policy)
+
+        Please note that in order to be able to execute this function, proper permissions should be set upfront for
+        the given user. To set permissions, do the following:
+        1. Enter to AWS console ==> "My Security Credentials" ==> Users ==> <relevant-user> ==> "Add inline policy"
+        2. set the following write permissions:
+            "iam:DetachRolePolicy", "iam:CreateRole", "iam:DeleteRole", "iam:AttachRolePolicy"
+        3. Provides a name to the new attached permissions and save
+        4. Make sure there are no any other contradict permission policy that block one of the permissions above.
```
Comment (liororama) [inline mlcomp/parallelm/ml_engine/sagemaker_engine.py]: contradict -> contradicting

File: mlcomp/parallelm/ml_engine/sagemaker_engine.py [hunk 2]
```diff
@@ -47,29 +50,63 @@ def iam_role(self):
...
+        """
+        now = datetime.datetime.now()
+        role_name = 'AmazonSageMaker-ExecutionRole-{y}{month}{d}T{h}{minute}{s}' \
+            .format(y=now.year, month=now.month, d=now.day, h=now.hour, minute=now.minute, s=now.second)
+
+        tags = [{'Key': self._tag_key, 'Value': self._tag_value}] if self._tag_key else []
+
+        trust_policy = {
+            "Version": "2012-10-17",
```
Comment (liororama) [inline mlcomp/parallelm/ml_engine/sagemaker_engine.py]: constant?

File: mlcomp/parallelm/ml_engine/sagemaker_engine.py [hunk 3]
```diff
@@ -47,29 +50,63 @@ def iam_role(self):
...
+
+            self._iam_role_name = response[SageMakerEngine.ROLE_RESPONSE_GROUP][SageMakerEngine.ROLE_RESPONSE_NAME_KEY]
+            self._iam_role = response[SageMakerEngine.ROLE_RESPONSE_GROUP][SageMakerEngine.ROLE_RESPONSE_ARN_KEY]
+
+            self._logger.info("Attaching sagemaker role to full access policy, name: {}, policy: {}"
+                              .format(role_name, SageMakerEngine.FULL_ACCESS_POLICY))
+            response = client.attach_role_policy(PolicyArn=SageMakerEngine.FULL_ACCESS_POLICY, RoleName=role_name)
+            self._logger.debug(pprint.pformat(response))
+        except ClientError as e:
+            self._logger.error("Failed to create a an iam role for sagemaker service!\n{}".format(e))
```
Comment (liororama) [inline mlcomp/parallelm/ml_engine/sagemaker_engine.py]: iam or IAM?

**Review by liororama — APPROVED** [review]

File: mlcomp/parallelm/ml_engine/sagemaker_engine.py (hunk 1 above)
Comment (zohar-mizrahi) [reply mlcomp/parallelm/ml_engine/sagemaker_engine.py]: Fixed

File: mlcomp/parallelm/ml_engine/sagemaker_engine.py (hunk 2 above)
Comment (zohar-mizrahi) [reply mlcomp/parallelm/ml_engine/sagemaker_engine.py]: Not really. This is a signature that comes from AWS SageMaker and expected to be seen in the given json as is.

File: mlcomp/parallelm/ml_engine/sagemaker_engine.py (hunk 3 above)
Comment (zohar-mizrahi) [reply mlcomp/parallelm/ml_engine/sagemaker_engine.py]: Fixed

**Pull request closed without merging**